icechunk-python/src/storage.rs

```diff
@@ -177,38 +168,21 @@ fn mk_credentials(config: Option<&PyS3Credentials>, anon: bool) -> S3Credentials
     }
 }
 
+impl From<PyStorageConfig> for StorageConfig {
+    fn from(storage: PyStorageConfig) -> Self {
+        storage.0
+    }
+}
+
 impl From<&PyStorageConfig> for StorageConfig {
     fn from(storage: &PyStorageConfig) -> Self {
-        match storage {
-            PyStorageConfig::Memory { prefix } => {
-                StorageConfig::InMemory { prefix: prefix.clone() }
-            }
-            PyStorageConfig::Filesystem { root } => {
-                StorageConfig::LocalFileSystem { root: PathBuf::from(root.clone()) }
-            }
-            PyStorageConfig::S3 {
-                bucket,
-                prefix,
-                anon,
-                credentials,
-                endpoint_url,
-                allow_http,
-                region,
-            } => {
-                let s3_config = S3Config {
-                    region: region.clone(),
-                    credentials: mk_credentials(credentials.as_ref(), *anon),
-                    endpoint: endpoint_url.clone(),
-                    allow_http: allow_http.unwrap_or(false),
-                };
+        storage.0.clone()
+    }
+}
 
-                StorageConfig::S3ObjectStore {
-                    bucket: bucket.clone(),
-                    prefix: prefix.clone(),
-                    config: Some(s3_config),
-                }
-            }
-        }
+impl AsRef<StorageConfig> for PyStorageConfig {
+    fn as_ref(&self) -> &StorageConfig {
+        &self.0
```

Comment:
This `AsRef` impl means that if you have a `PyStorageConfig` and you want to access an `&StorageConfig`, you can just call `.as_ref()`.